enclave/src/attestation.rs

```diff
@@ -108,36 +88,25 @@ pub unsafe extern "C" fn get_mrenclave(mrenclave: *mut u8, mrenclave_size: u32)
 }
 
 pub fn get_mrenclave_of_self() -> SgxResult<sgx_measurement_t> {
-    Ok(get_report_of_self()?.mr_enclave)
+    let ocall_api = OCallComponentFactory::get_attestation_api();
+
+    Ok(get_report_of_self(ocall_api)?.mr_enclave)
 }
 
-fn get_report_of_self() -> SgxResult<sgx_report_body_t> {
+fn get_report_of_self<A: EnclaveAttestationOCallApi>(
+    ocall_api: Arc<A>,
+) -> SgxResult<sgx_report_body_t> {
```

Comment:
Functions like this one can now use the abstraction, provided by the `EclaveAttestationOCallApi`